### PR TITLE
Fix - accepting GitHub factories may ignore devfile.yaml in repository.

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
When added support for Bitbucket, we're accidentally broken the search of devfile in the repository for GitHub.
This PR fixes the situation.

### How to test this PR?

Current master ignores devfile in the repository and always apply default params. Atfer PR, actual devfile is applied.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by metlos — COMMENTED** [review]
Wouldn't it be better for the `GithubUrl.rawFileLocation(loc)` to just return the `loc` unmodified if it saw it was an absolute URL? That's what `URI.resolve()` does I think and so would be a kind of expected behavior.

**Comment by mshaposhnik** [comment]
This logic is required for retrieving relative content and explicitly described in FileContentProvider interface.  Moving it into each URL object implementation (GitHub, Bitbuket, Default) imo will make it kind of hidden and not so explicit.

**Review by sparkoo — APPROVED** [review]
few nitpicks

File: wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubFileContentProvider.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import java.net.URISyntaxException;
+import org.eclipse.che.api.workspace.server.devfile.FileContentProvider;
+import org.eclipse.che.api.workspace.server.devfile.URLFetcher;
+import org.eclipse.che.api.workspace.server.devfile.exception.DevfileException;
+
+/** Github specific file content provider. */
+class GithubFileContentProvider implements FileContentProvider {
+
+  private final GithubUrl githubUrl;
+  private URLFetcher urlFetcher;
```
[the remark this answers is not in the document]
Comment (mshaposhnik) [reply wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubFileContentProvider.java]: \+

**Comment by metlos** [comment]
> This logic is required for retrieving relative content and explicitly described in FileContentProvider interface. Moving it into each URL object implementation (GitHub, Bitbuket, Default) imo will make it kind of hidden and not so expected/explicit.

`FileContentProvider#fetchContent(fileUrl)` specifies only that `@param fileURL absolute or devfile-relative file URL to fetch content`, so the individual impls kinda ARE left to their own devices to fulfill that..

**Review by skabashnyuk — APPROVED** [review]

**Pull request merged**